firedrake/functionspaceimpl.py

```diff
@@ -815,6 +903,7 @@ def __init__(self, spaces, name=None):
         self._ufl_function_space = ufl.FunctionSpace(mesh.ufl_mesh(),
                                                      finat.ufl.MixedElement(*[s.ufl_element() for s in spaces]))
         self.name = name or "_".join(str(s.name) for s in spaces)
+        self._label = "_".join(str(s._label) for s in spaces)
```

Comment:
Probably:
```suggestion
        label = ""
        for s in spaces:
            label += "(" + s._label + ")_"
        self._label = label
```
or anything that is not ambiguous.